src/platform/test.rs

```diff
@@ -412,6 +414,47 @@ fn receiver_set() {
 }
 
 #[test]
+#[cfg(not(any(feature = "force-inprocess", target_os = "windows", target_os = "android")))]
+fn receiver_set_eintr() {
+    let (start_server, start_name) = OsIpcOneShotServer::new().unwrap();
+    let (end_server, end_name) = OsIpcOneShotServer::new().unwrap();
```

Comment:
> Why are you using a separate server (channel) for the final result, rather than just sending the success message over the first channel as well?

Sorry, just hadn't used `OneShotServers` before and didn't read the code close enough. Updated in the latest changes.